src/App/AppEffects.h

```diff
@@ -223,39 +224,45 @@ class Frizzles2D:public Effect {
   }
 }; // Frizzles2D
 
+#define NUM_GEQ_CHANNELS 16                                           // number of frequency channels. Don't change !!
+
 class GEQEffect:public Effect {
 public:
+  byte previousBarHeight[1024];
+  uint8_t intensity;
+  uint8_t speed;
+  bool check1;
+  bool check2;
+
   const char * name() {
     return "GEQ";
   }
-  void setup() {} //not implemented yet
+
+  void setup() {
+    fadeToBlackBy( ledsP, LedsV::nrOfLedsP, 16);
+    for (int i=0; i<LedsV::widthV; i++) previousBarHeight[i] = 0;
+  } //not implemented yet
+
   void loop() {
-    if (!strip.isMatrix) return mode_static(); // not a 2D set-up
 
     const int NUM_BANDS = NUM_GEQ_CHANNELS ; // map(SEGMENT.custom1, 0, 255, 1, 16);
-    const uint16_t cols = LedsV::widthV 
-    const uint16_t rows = LedsV::heightV 
-    if ((cols <=1) || (rows <=1)) return mode_static(); // not really a 2D set-up
+    const uint16_t cols = LedsV::widthV;
+    const uint16_t rows = LedsV::heightV; 
 
-    if (!SEGENV.allocateData(cols*sizeof(uint16_t))) return mode_static(); //allocation failed
-    uint16_t *previousBarHeight = reinterpret_cast<uint16_t*>(SEGENV.data); //array of previous bar heights per frequency band
 
     uint8_t *fftResult = wledAudioMod->fftResults;
     #ifdef SR_DEBUG
     uint8_t samplePeak = *(uint8_t*)um_data->u_data[3];
     #endif
 
-    if (SEGENV.call == 0) for (int i=0; i<cols; i++) previousBarHeight[i] = 0;
-
     bool rippleTime = false;
-    if (millis() - SEGENV.step >= (256U - SEGMENT.intensity)) {
-      SEGENV.step = millis();
+    if (millis() - step >= (256U - intensity)) {
+      step = millis();
       rippleTime = true;
     }
 
-    if (SEGENV.call == 0) SEGMENT.fill(BLACK);
-    int fadeoutDelay = (256 - SEGMENT.speed) / 64;
-    if ((fadeoutDelay <= 1 ) || ((SEGENV.call % fadeoutDelay) == 0)) SEGMENT.fadeToBlackBy(SEGMENT.speed);
```

Comment:
To make porting easier and generally code nicer - should we not just a have  `fadeToBlackBy` method that takes only speed (assuming you are controlling all the current LEDs)

Reply:
Maybe, don't know yet, I now converted them to the fastled version, which I like also (the fastled compatibility)